rust/noosphere-gateway/src/gateway.rs

```diff
@@ -85,17 +86,31 @@ where
     );
 
     let app = Router::new()
-        .route(&GatewayRoute::Did.to_string(), get(did_route))
         .route(
-            &GatewayRoute::Replicate(None).to_string(),
-            get(replicate_route::<C, S>),
+            &v0alpha1::Route::Did.to_string(),
+            get(handlers::v0alpha1::did_route),
         )
         .route(
-            &GatewayRoute::Identify.to_string(),
-            get(identify_route::<C, S>),
+            &v0alpha1::Route::Replicate(None).to_string(),
+            get(handlers::v0alpha1::replicate_route::<C, S>),
+        )
+        .route(
+            &v0alpha1::Route::Identify.to_string(),
+            get(handlers::v0alpha1::identify_route::<C, S>),
+        )
+        .route(
+            &v0alpha1::Route::Push.to_string(),
+            #[allow(deprecated)]
+            put(handlers::v0alpha1::push_route::<C, S>),
+        )
+        .route(
+            &v0alpha2::Route::Push.to_string(),
+            put(handlers::v0alpha2::push_route::<C, S>),
+        )
+        .route(
+            &v0alpha1::Route::Fetch.to_string(),
```

Comment:
Mixing `v0alpha1` and `v0alpha2` here (outside of the deprecated routes maybe) feels like a smell; could `v0alpha2` re-export `v0alpha1` routes where there is no change?

Reply:
I guess it's a question of style, but the distinction is useful because we mount *both* `v0alpha1` and `v0alpha2` routes (see a few lines above) for push, which let's us enable clients to upgrade without breaking in-between.

Reply:
Backwards compat is for sure a larger topic to tackle, I need to think about this more